### PR TITLE
[circle-quantizer] Extract QConfReader

### PR DESCRIPTION
**Pull request by seanshpark**
This will extract QConfReader for reading qconf json file.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
for #11280 , from draft  #11513

**Comment by seanshpark** [comment]
@jinevening ping

File: compiler/circle-quantizer/src/CircleQuantizer.cpp [hunk 1]
```diff
@@ -38,8 +38,40 @@ using LayerParams = luci::CircleQuantizer::Options::LayerParams;
 using Algorithms = luci::CircleQuantizer::Options::Algorithm;
 using AlgorithmParameters = luci::CircleQuantizer::Options::AlgorithmParameters;
 
+struct QConfReader
+{
+  void init(std::string &filename)
```
Comment (jinevening) [inline compiler/circle-quantizer/src/CircleQuantizer.cpp:43]:
```suggestion
  void init(const std::string &filename)
```

File: compiler/circle-quantizer/src/CircleQuantizer.cpp [hunk 2]
```diff
@@ -38,8 +38,40 @@ using LayerParams = luci::CircleQuantizer::Options::LayerParams;
...
+    // Failed to open cfg file
+    if (not _ifs.is_open())
+      throw std::runtime_error("Cannot open config file. " + filename);
+
+    JSONCPP_STRING errs;
+    Json::CharReaderBuilder builder;
+
+    // Failed to parse
+    if (not parseFromStream(builder, _ifs, &_root, &errs))
+      throw std::runtime_error("Cannot parse config file (json format). " + errs);
```
Comment (jinevening) [inline compiler/circle-quantizer/src/CircleQuantizer.cpp:56]: It would be better to close `_ifs` right after parsing, rather than in destructor.

**Review by jinevening — APPROVED** [review]
LGTM Thank you!

**Pull request merged**